src/Microsoft.Azure.WebJobs.Logging.ApplicationInsights/ApplicationInsightsLogger.cs

```diff
@@ -426,9 +429,9 @@ private void StartTelemetryIfFunctionInvocation(IDictionary<string, object> stat
                 return;
             }
 
-            // Http and ServiceBus triggers are tracked automatically by the ApplicationInsights SDK
+            // HTTP and ServiceBus triggers are tracked automatically by the ApplicationInsights SDK
             // In such case a current Activity is present.
-            // We won't track and only stamp function specific details on the RequestTelemtery
+            // We won't track and only stamp function specific details on the RequestTelemetry
             // created by SDK via Activity when function ends
             if (Activity.Current == null)
```

Comment:
We'll still need to do something like:
```
if (Activity.Current == null ||
   DictionaryLoggerScope.GetMergedStateDictionary().ContainsKey("MS_IgnoreActivity"))
```

So that we can skip portal invocations. I'm going to do this locally so I can test fully E2E.

Reply:
Did we need that because portal calls are not tracked atuomagically and we need to track them explicitly? I remember you mentioned it was fixed or I missed something.  Was it done with  'MS_IgnoreActivity'? If this workaround is needed, I'd love to understand why.

Let's return MS_IgnoreActivity for now and I'll check what happens with portal invocations separately.

Reply:
When we invoke (for example) a QueueTrigger via the portal, we do that with an Http request. And, more importantly, we actually invoke the function on a background Task and immediately return 202. 

So with the auto-Http tracking, App Insights sees that very fast response as the function execution and records it. When, in reality, we want to ignore that Http request and do the "non-Http" tracking pipeline for that invocation.

We'll need to continue doing this -- for a while it was broken b/c of something unrelated to Http tracking. All of my tests were failing b/c we weren't properly giving the "Executed..." TraceTelemetry the correct operation details. So it wouldn't show up. I fixed that in https://github.com/Azure/azure-webjobs-sdk/pull/2058. 

I've done some validation this afternoon with the "MS_IgnoreActivity" change and things are looking good. I'm setting up a private site extension now with some different trigger types and want to run through some more scenarios, but so far, so good.

Reply:
thanks a lot for the explanation, it makes sense